src/kOS.Safe/Encapsulation/Structure.cs

```diff
@@ -337,20 +337,10 @@ public static Structure FromPrimitiveWithAssert(object value)
 
         public static object ToPrimitive(object value)
         {
-            var scalarValue = value as ScalarValue;
-            if (scalarValue != null)
+            var primitive = value as PrimitiveStructure;
+            if (primitive != null)
             {
-                return scalarValue.Value;
-            }
-            var booleanValue = value as BooleanValue;
-            if (booleanValue != null)
-            {
-                return booleanValue.Value;
-            }
-            var stringValue = value as StringValue;
-            if (stringValue != null)
-            {
-                return stringValue.ToString();
+                return primitive.ToPrimitive();
```

Comment:
No, I don't really have any work with it.  It's just a matter of wishing we had a better way to interface this.  I don't like having half of the feature residing on the Sub-class while the remainder resides on the base class.  It means having to remember 2 different locations if we need to change the logic, rather than keeping it in one place.  The only way to get the conversion from a `double` to a `ScalarValue` to be implemented on the `double` object is to use an extension method or reflection.  Neither of which work well on an un-cast `object` type.  

I'm also used to a serialization structure that gets annotated onto classes using attributes instead of by inheritance and interfaces.  So any time we move a "core" function to make serialization easier it seems a little weird to me.  But it will function fine.

So I guess I'd punt.  No changes necessary on my end.  Maybe a comment in the `FromPrimitive` method to point to the `PrimitiveStructure` implementation.  But that isn't much of a benefit either.